torch/csrc/jit/passes/dead_code_elimination.cpp

```diff
@@ -227,18 +227,49 @@ class DeadCodeEliminator {
     return has_side_effects;
   }
 
+  void removeIfNodeOutput(Node * node, size_t i) {
+    node->eraseOutput(i);
+    for (Block* b : node->blocks()) {
+      b->eraseOutput(i);
+    }
+  }
+
   void removeDeadBlockOutputs(Node* node) {
     if (node->kind() != prim::If && node->kind() != prim::GradOf) {
       return;
     }
 
     for (size_t i_1 = node->outputs().size(); i_1 > 0; --i_1) {
       size_t i = i_1 - 1;
-      if (!node->outputs().at(i)->hasUses()) {
-        node->eraseOutput(i);
-        for (Block* b : node->blocks()) {
-          b->eraseOutput(i);
+      if (node->kind() == prim::If) {
+        Value* b1_val = node->blocks().at(0)->outputs().at(i);
```

Comment:
Can you refactor the unchecked_unwrap_optional part into a meaningfully named function?

```
if (removePossiblyUnneededUnwrap(...))
  continue;
```